moveos/moveos-types/src/event.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::ensure;
+use std::str::FromStr;
+// use move_bytecode_utils::module_cache::GetModule;
+use move_core_types::account_address::AccountAddress;
+// use move_core_types::identifier::IdentStr;
+// use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::StructTag;
+// use move_core_types::value::MoveStruct;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use serde_with::serde_as;
+use serde_with::Bytes;
+
+// use crate::base_types::{ObjectID, AccountAddress, H256};
+// use crate::{H256};
+use crate::h256::H256;
+
+/// A universal Rooch event type encapsulating different types of events
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EventEnvelope {
+    /// Transaction Info Id
+    pub tx_info_id: H256,
+    /// Consecutive per-tx counter assigned to this event.
+    pub event_num: u64,
+    /// Specific event type
+    pub event: Event,
+    /// Move event's json value
+    pub parsed_json: Value,
+    /// UTC timestamp in milliseconds since epoch (1/1/1970)
+    pub timestamp: u64,
+}
+/// Unique ID of a Rooch Event, the ID is a combination of tx info id and event seq number,
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EventID {
+    pub tx_info_id: H256,
```

Comment:
This id should be the `tx_id`, because the merkle root of events in the TransactionInfo, we can not get the `tx_info_id` before init the TransactionInfo.

Reply:
fixed